protostar/compiler/project_compiler.py

```diff
@@ -29,42 +48,55 @@ def __init__(
         self._project_root_path = project_root_path
         self.configuration_file = configuration_file
 
-    def compile_contract_to_sierra_from_contract_identifier(
-        self, contract_identifier: ContractIdentifier
-    ) -> str:
-        return self._compile_cairo1_contract_from_contract_identifier(
-            contract_identifier=contract_identifier, output_type=_OutputType.SIERRA
-        )
+        self._sierra_file_extension = ".sierra.json"
+        self._casm_file_extension = ".casm.json"
 
-    def compile_contract_to_sierra_from_contract_name(self, contract_name: str) -> str:
-        return self._compile_cairo1_contract_from_contract_name(
-            contract_name=contract_name, output_type=_OutputType.SIERRA
+    def compile_contract(
+        self,
+        contract_identifier: ContractIdentifier,
+        cairo_path: Optional[list[Path]] = None,
+        output_path: Optional[Path] = None,
```

Comment:
these repetitive args look like some boilerplate, maybe we could replace them with a data class or something?

Reply:
i disagree on this one, not that many args here, it would make things less readable

Reply:
I also don't think there are too many args here. Grouping them would be "artificial"

Reply:
ok, so be it